src/execution/compiler/operator/sort_translator.cpp

```diff
@@ -233,6 +249,20 @@ void SortTranslator::FinishPipelineWork(const Pipeline &pipeline, FunctionBuilde
     } else {
       function->Append(codegen->SorterSort(sorter_ptr));
     }
+
+    FeatureRecord(function, brain::ExecutionOperatingUnitType::SORT_BUILD,
+                  brain::ExecutionOperatingUnitFeatureAttribute::NUM_ROWS, pipeline,
+                  num_sort_build_rows_.Get(GetCodeGen()));
+    FeatureRecord(function, brain::ExecutionOperatingUnitType::SORT_BUILD,
+                  brain::ExecutionOperatingUnitFeatureAttribute::CARDINALITY, pipeline,
+                  codegen->CallBuiltin(ast::Builtin::SorterGetTupleCount, {sorter_ptr}));
```

Comment:
Since we don't necessarily want to track data uniqueness, this might be the best we can do? (although the prediction results will be slightly affected since `sort_build` does depend on the input data cardinality).

@malin1993ml

Reply:
Yeah I think this is the best we can do. I would add a TODO here to document that in theory we should use the number of unique tuples as cardinality, but given the potential overhead and engineering complexity we just use the size of the sorter.

Reply:
I was also running TPCC just now with this branch. I noticed that this value as well as the cardinality value for `SORT_ITERATE` are recorded as a very high corrupted value. This probably needs double-check.

Reply:
It works for small test cases for me.. Does it go above int32_max? I wonder if we are hitting #1049.

Reply:
I don't think it goes above int32_max. I just directly compiled your branch and executed TPCC against it with `--metrics_pipeline` turned on. Maybe try TPCC locally to see whether you can reproduce?

Reply:
Fixed, it was a bug with bytecode generation.